optax/_src/transform.py

```diff
@@ -869,6 +869,64 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByNewtonState(NamedTuple):
+  """State holding the inverse of the sum of gradient squares to date."""
+
+  hessian_inv: base.Updates
+
+
+def sherman_morrison(a_inv, u, v):
+  den = 1.0 + (u @ a_inv) @ v
+  a_inv -= a_inv @ jnp.outer(u, v) @ a_inv / den
+  return a_inv
+
+
+def scale_by_newton(eps: float = 1e-7) -> base.GradientTransformation:
+  """Rescale updates by multiplying by the inverse of an approximation of the
+  hessian.
+
+  References
+  ----------
+  [^1]: [Hazan, E., Agarwal, A. and Kale, S., 2007. Logarithmic regret
+  algorithms for online convex optimization. Machine Learning, 69(2-3),
+  pp.169-192](https://www.cs.princeton.edu/~ehazan/papers/log-journal.pdf)
```

Comment:
Could you add which algorithm/formula/section it is in the paper so that it's easier for the reader to cross-reference?

Reply:
Yes the correct link is this one: https://link.springer.com/content/pdf/10.1007/s10994-007-5016-8.pdf